data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineConditionalRoute.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.model.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Model representing a route within a pipeline.
+ *
+ * @since 2.0
+ */
+@JsonSerialize(using = PipelineConditionalRoute.PipelineConditionalRouteSerializer.class)
+@JsonDeserialize(using = PipelineConditionalRoute.PipelineConditionalRouteDeserializer.class)
+public class PipelineConditionalRoute {
+    private final String name;
+    private final String condition;
```

Comment:
Should we think about an interface for condition ? This will help us do some validation over condition.

Reply:
I thought about this, but I really think it should be a larger change that covers any Data Prepper condition (or even expression). It thus affects the data-prepper-expression package and would be a fairly significant change.